pdp-server/src/main.rs

```diff
@@ -0,0 +1,202 @@
+mod api;
+mod cache;
+mod config;
+mod errors;
+mod headers;
+mod models;
+mod openapi;
+mod state;
+
+use crate::state::AppState;
+use axum::Router;
+use log::{error, info};
+use std::net::SocketAddr;
+use utoipa::OpenApi;
+use utoipa_axum::router::OpenApiRouter;
+use utoipa_scalar::{Scalar, Servable};
+
+#[tokio::main]
+async fn main() {
+    // Initialize logging
+    env_logger::init_from_env(env_logger::Env::default().default_filter_or("info"));
+
+    // Load configuration
+    let settings = match config::Settings::new() {
+        Ok(settings) => settings,
+        Err(e) => {
+            error!("Configuration error: {}", e);
+            std::process::exit(1);
+        }
+    };
+
+    // Initialize cache asynchronously without using block_in_place
+    let cache = match cache::create_cache(&settings).await {
+        Ok(cache) => cache,
+        Err(e) => {
+            error!("Failed to initialize cache: {}", e);
+            std::process::exit(1);
+        }
+    };
+
+    // Initialize application state
+    let state: AppState = AppState::new_python_based(settings.clone(), cache)
+        .await
+        .expect("Failed to initialize application state");
+
+    // Create application & Initialize PDPEngine
+    let app = create_app(state.clone()).await;
```

Comment:
Isn't there any "on statup" and "on shutdown" for that we can utilize here instead of starting the pdp engine in the create app and stopping it when the app finishes ?